host-bin/edit-chroot

```diff
@@ -385,8 +432,22 @@ for NAME in "$@"; do
         fi
         echo -n "  Restoring $src to $CHROOT..." 1>&2
         mkdir -p "$CHROOT"
+        if [ -f "$src.part01" ]; then
+            tarsrc="`mktemp -u --tmpdir=/tmp 'crouton-restore.XXX'`"
+            mkfifo -m 600 "$tarsrc"
+            # Pipe through dd to eat the EOFs
```

Comment:
Okay, managed to reproduce with small split sizes. `tar` does not like partial reads when reading from what it thinks is a file (and when no compression is used). You should add `-B` to the extract operation below:

```
-B, --read-full-records
              When listing or extracting, accept incomplete input records  after  end-of-
              file marker
```
(I don't think the help text makes too much sense, I was always told that partial reads do not mean EOF (0-length read does); it's clearer when you look at `tar` source)

Note that using `cat fifo | tar xvf -` sets the option by default, as well as when the archive is compressed (since `gunzip` writes irregular blocks).

The `dd` hack only works because more data gets buffered (dd block size is 512, while tar blocks appear to be 10k, so it's not guaranteed to work).